src/layers/PassagierfrequenzenLayer/PassagierfrequenzenLayer.js

```diff
@@ -0,0 +1,184 @@
+import qs from 'querystring';
+import VectorLayer from 'react-spatial/layers/VectorLayer';
+import OLVectorLayer from 'ol/layer/Vector';
+import OLVectorSource from 'ol/source/Vector';
+import { unByKey } from 'ol/Observable';
+import { Style, Stroke, Circle, Fill } from 'ol/style';
+import OLGeoJSON from 'ol/format/GeoJSON';
+import CONF from '../../config/appConfig';
+import layerHelper from '../layerHelper';
+
+class PassagierfrequenzenLayer extends VectorLayer {
+  constructor(options = {}) {
+    const name = 'ch.sbb.bahnhoffrequenzen';
+    const key = 'ch.sbb.bahnhoffrequenzen';
+
+    const olLayer = new OLVectorLayer({
+      source: new OLVectorSource(),
+    });
+
+    super({
+      ...options,
+      name,
+      key,
+      olLayer,
+    });
+
+    this.styleCache = {};
+
+    this.url = `${CONF.geoserverUrl}?`;
+    this.urlParams = {
+      service: 'WFS',
+      version: '1.0.0',
+      request: 'GetFeature',
+      typeName: 'trafimage:passagierfrequenzen',
+    };
+    this.olLayer.setStyle((f, r) => this.styleFunction(f, r));
+    this.olLayer.getSource().setLoader(this.loader.bind(this));
+
+    this.onClick(f => {
+      const [clickedFeature] = f;
+      this.clickedFeatureName = clickedFeature && clickedFeature.get('name');
```

Comment:
Please use the feature itself instead of its name.

Reply:
There is a feature per resolution, the comparison feature against feature will fail if I zoom in. Therefore the highlighting only works for the same resolution, and disappear when zooming. Using the name is the solution to ensure the highlighting works when zooming